sirepo/pkcli/job_cmd.py

```diff
@@ -148,9 +148,9 @@ def _do_download_data_file(msg, template):
             c = pkcompat.to_bytes(pkio.read_text(r.filename)) \
                 if u.endswith(('.py', '.txt', '.csv')) \
                 else r.filename.read_binary()
-            c, b = _change_message_too_big(c)
-            if b:
-                return c
+            c = _validate_and_jsonl(c)
+            if 'error' in pkjson.load_any(c):
```

Comment:
This is not going to work, and this is going to be very expensive (load_any)

Two functions are better, because they document what's going on. Private functions are "cheap" from a cognitive load and don't incur a cost by being a public function.

Reply:
Cool. I'll do it with two

Reply:
I wasn't clear about what "going to work" means. Looking for `error` in a string is a fine thing to do if one is parsing, say, the output of a code. However, in this case, there is an arbitrary result coming back, and we are looking for a specific case. You could write it as:

```py
if 'error' == pkjson.load_any(c).get('state'):
```

However, this makes an assumption, the value of `c` is a `dict` (or PKDict). That's the type of assumption that gets us into trouble in software, and is what I call an "implicit coupling". A safer test would be:

```py
x = pkjson.load_any(c)
if isinstance(x, dict) and 'error' == x.get('state'):
```

However, this is increasing the cognitive load of the whole problem, and it would be clearer to say:

```py
if len(c) >= job.cfg.max_message_bytes:
    return json....PKDict(state='error'...))
```

The problem with this is that it is duplicating code unnecessarily. We already have the concept named, and it can be "explicitly coupled" via a separate function. That's what refactoring is all about: improving the state of existing code. 

That's why two functions are better than other solutions. You factor out a concept from a clearly named function without changing what it does.